ops: Seeing failures for offline request of metosin reitit

Reitit has submodules. I must've missed something.

Exception from local repro:
```
clojure.lang.ExceptionInfo: java.lang.NullPointerException in Interceptor :cljdoc.server.pedestal/offline-docset - Cannot invoke "String.substring(int)" because "s" is null {:execution-id nil, :stage :enter, :interceptor :cljdoc.server.pedestal/offline-docset, :exception-type :java.lang.NullPointerException}
	at io.pedestal.interceptor.chain$throwable__GT_ex_info.invokeStatic(chain.clj:42)
	at io.pedestal.interceptor.chain$throwable__GT_ex_info.invoke(chain.clj:36)
	at io.pedestal.interceptor.chain$begin_error.invokeStatic(chain.clj:74)
	at io.pedestal.interceptor.chain$begin_error.invoke(chain.clj:62)
	at io.pedestal.interceptor.chain$try_stage.invokeStatic(chain.clj:124)
	at io.pedestal.interceptor.chain$try_stage.invoke(chain.clj:108)
	at io.pedestal.interceptor.chain$execute_enter$fn__13823.invoke(chain.clj:232)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1990)
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1990)
	at clojure.lang.RestFn.invoke(RestFn.java:428)
	at io.pedestal.interceptor.chain$execute_enter.invokeStatic(chain.clj:212)
	at io.pedestal.interceptor.chain$execute_enter.invoke(chain.clj:202)
	at io.pedestal.interceptor.chain$execute_continue.invokeStatic(chain.clj:363)
	at io.pedestal.interceptor.chain$execute_continue.invoke(chain.clj:360)
	at io.pedestal.interceptor.chain$execute.invokeStatic(chain.clj:425)
	at io.pedestal.interceptor.chain$execute.invoke(chain.clj:374)
	at io.pedestal.interceptor.chain$execute.invokeStatic(chain.clj:430)
	at io.pedestal.interceptor.chain$execute.invoke(chain.clj:374)
	at io.pedestal.http.impl.servlet_interceptor$interceptor_service_fn$fn__45128.invoke(servlet_interceptor.clj:386)
	at io.pedestal.servlet.FnServlet.service(FnServlet.java:57)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
	at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:751)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1622)
	at org.eclipse.jetty.ee10.websocket.servlet.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:195)
	at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1594)
	at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1555)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
	at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:439)
	at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:470)
	at org.eclipse.jetty.ee10.servlet.SessionHandler.handle(SessionHandler.java:717)
	at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1096)
	at org.eclipse.jetty.server.Server.handle(Server.java:182)
	at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:677)
	at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:416)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:492)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.epcRunTask(AdaptiveExecutionStrategy.java:428)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:401)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:255)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:204)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:981)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1211)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1166)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.NullPointerException: Cannot invoke "String.substring(int)" because "s" is null
	at clojure.core$subs.invokeStatic(core.clj:5039)
	at clojure.core$subs.invoke(core.clj:5034)
	at io.pedestal.http.route.sawtooth.impl$match_by_path$match_on_path__14826.invoke(impl.clj:333)
	at io.pedestal.http.route.sawtooth.impl$subdivide_by_request_key$match_request_key__14844.invoke(impl.clj:385)
	at io.pedestal.http.route.sawtooth$router$fn__14879.invoke(sawtooth.clj:28)
	at io.pedestal.http.route$route_context.invokeStatic(route.clj:448)
	at io.pedestal.http.route$route_context.invoke(route.clj:446)
	at io.pedestal.http.route$construct_router_interceptor_from_table$fn__15066.invoke(route.clj:470)
	at io.pedestal.http.route$try_routing_for.invokeStatic(route.clj:637)
	at io.pedestal.http.route$try_routing_for.invoke(route.clj:618)
	at io.pedestal.http.route$try_routing_for.invokeStatic(route.clj:631)
	at io.pedestal.http.route$try_routing_for.invoke(route.clj:618)
	at cljdoc.server.routes$match_route.invokeStatic(routes.clj:126)
	at cljdoc.server.routes$match_route.invoke(routes.clj:125)
	at cljdoc.util.fixref$fix.invokeStatic(fixref.clj:165)
	at cljdoc.util.fixref$fix.invoke(fixref.clj:110)
	at cljdoc.render.offline$doc_page.invokeStatic(offline.clj:123)
	at cljdoc.render.offline$doc_page.invoke(offline.clj:118)
	at cljdoc.render.offline$docs_files$iter__29738__29742$fn__29743.invoke(offline.clj:196)
	at clojure.lang.LazySeq.force(LazySeq.java:50)
	at clojure.lang.LazySeq.realize(LazySeq.java:87)
	at clojure.lang.LazySeq.seq(LazySeq.java:104)
	at clojure.lang.Cons.next(Cons.java:41)
	at clojure.lang.RT.next(RT.java:733)
	at clojure.core$next__5472.invokeStatic(core.clj:64)
	at clojure.core.protocols$fn__8279.invokeStatic(protocols.clj:168)
	at clojure.core.protocols$fn__8279.invoke(protocols.clj:123)
	at clojure.core.protocols$fn__8233$G__8228__8242.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8266.invokeStatic(protocols.clj:74)
	at clojure.core.protocols$fn__8266.invoke(protocols.clj:74)
	at clojure.core.protocols$fn__8207$G__8202__8220.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6965)
	at clojure.core$into.invokeStatic(core.clj:7038)
	at clojure.core$into.invoke(core.clj:7029)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:404)
	at clojure.core$reduce.invokeStatic(core.clj:6960)
	at clojure.core$reduce.invoke(core.clj:6947)
	at cljdoc.render.offline$docs_files.invokeStatic(offline.clj:172)
	at cljdoc.render.offline$docs_files.invoke(offline.clj:136)
	at cljdoc.render.offline$zip_stream.invokeStatic(offline.clj:222)
	at cljdoc.render.offline$zip_stream.invoke(offline.clj:218)
	at cljdoc.server.pedestal$offline_docset__45929.invokeStatic(pedestal.clj:535)
	at cljdoc.server.pedestal$offline_docset__45929.invoke(pedestal.clj:526)
	at io.pedestal.interceptor.chain$try_stage.invokeStatic(chain.clj:113)
	... 46 common frames omitted
```

--- a/src/cljdoc/util/fixref.clj
+++ b/src/cljdoc/util/fixref.clj
@@ -159,7 +159,10 @@
                                            (map (fn [^Element e] (.attributes e)))
                                            (filter (fn [^Attributes a] (absolute-uri? (.get a "href")))))]
       (let [href (.get absolute-link "href")
-            {:keys [host scheme ^String path] :as uri} (uri/parse href)]
+            {:keys [host scheme ^String path] :as uri} (let [uri (uri/parse href)]
+                                                         (if (:path uri)
+                                                           uri
+                                                           (assoc uri :path "/")))]
         (if-not (and (= "https" scheme) (= "cljdoc.org" host))
           (.put absolute-link "rel" "nofollow")
           (let [{:keys [route-name path-params]} (routes/match-route path)

--- a/test/cljdoc/util/fixref_test.clj
+++ b/test/cljdoc/util/fixref_test.clj
@@ -80,9 +80,9 @@
   (t/is
    (match?
     (m/nested-equals
-      [[:a {:href "/some/path/here"} "absolute link to cljdoc"]
-       [:a {:href "/"} "root path"]
-       [:a {:href "/"} "no path"]])
+     [[:a {:href "/some/path/here"} "absolute link to cljdoc"]
+      [:a {:href "/"} "root path"]
+      [:a {:href "/"} "no path"]])
     (fix [[:a {:href "https://cljdoc.org/some/path/here"} "absolute link to cljdoc"]
           [:a {:href "https://cljdoc.org/"} "root path"]
           [:a {:href "https://cljdoc.org"} "no path"]]

--- a/test/cljdoc/util/fixref_test.clj
+++ b/test/cljdoc/util/fixref_test.clj
@@ -80,8 +80,12 @@
   (t/is
    (match?
     (m/nested-equals
-     [[:a {:href "/some/path/here"} "absolute link to cljdoc"]])
-    (fix [[:a {:href "https://cljdoc.org/some/path/here"} "absolute link to cljdoc"]]
+      [[:a {:href "/some/path/here"} "absolute link to cljdoc"]
+       [:a {:href "/"} "root path"]
+       [:a {:href "/"} "no path"]])
+    (fix [[:a {:href "https://cljdoc.org/some/path/here"} "absolute link to cljdoc"]
+          [:a {:href "https://cljdoc.org/"} "root path"]
+          [:a {:href "https://cljdoc.org"} "no path"]]
          fix-opts))))
 
 (t/deftest unknown-scm-links-when-relative-point-to-normalized-scm-test
@@ -278,7 +282,10 @@
                  [:a {:href "https://cljdoc.org/d/not-mygroupid/not-myartifactid/CURRENT"}
                   "doesn't replace CURRENT if not my group-id/artifact-id"]
                  [:a {:href "https://cljdoc.org/doopsie/mygroupid/myartifactid/CURRENT"}
-                  "doesn't replace CURRENT if not a cljdoc route"]]]
+                  "doesn't replace CURRENT if not a cljdoc route"]
+                 [:a {:href "https://cljdoc.org/some/path"} "arbitrary path"]
+                 [:a {:href "https://cljdoc.org/"} "root path"]
+                 [:a {:href "https://cljdoc.org"} "no path"]]]
       (t/is
        (match?
         (m/nested-equals input)

